### PR TITLE
Use pattern matching in arrow test utils to improve readability

### PR DESCRIPTION
**Pull request by SARAVANA1501**
Improves the readability using c# pattern matching
#25

### CODE REVIEW

File: src/csharp/Microsoft.Spark.UnitTest/TestUtils/ArrowTestUtils.cs [hunk 1]
```diff
@@ -30,82 +30,29 @@ public static void AssertEquals(string expectedValue, DataFrameColumn arrowArray
...
-            }
-
-            if (typeof(T) == typeof(byte[]))
-            {
-                return BinaryType.Default;
-            }
-
-            throw new NotSupportedException($"Unknown type: {typeof(T)}");
+            Type type = typeof(T);
+            IArrowType arrowType = type switch
```
Comment (suhsteve) [inline src/csharp/Microsoft.Spark.UnitTest/TestUtils/ArrowTestUtils.cs:34]:
```suggestion
            return type switch
```

File: src/csharp/Microsoft.Spark.UnitTest/TestUtils/ArrowTestUtils.cs [hunk 2]
```diff
@@ -30,82 +30,29 @@ public static void AssertEquals(string expectedValue, DataFrameColumn arrowArray
...
+                _ when type == typeof(uint) => UInt32Type.Default,
+                _ when type == typeof(long) => Int64Type.Default,
+                _ when type == typeof(ulong) => UInt64Type.Default,
+                _ when type == typeof(float) => FloatType.Default,
+                _ when type == typeof(double) => DoubleType.Default,
+                _ when type == typeof(DateTime) => Date64Type.Default,
+                _ when type == typeof(TimeSpan) => TimestampType.Default,
+                _ when type == typeof(string) => StringType.Default,
+                _ when type == typeof(byte[]) => BinaryType.Default,
+                _ => null
```
Comment (suhsteve) [inline src/csharp/Microsoft.Spark.UnitTest/TestUtils/ArrowTestUtils.cs:51]:
```suggestion
                _ => throw new NotSupportedException($"Unknown type: {typeof(T)}")
```

File: src/csharp/Microsoft.Spark.UnitTest/TestUtils/ArrowTestUtils.cs [hunk 3]
```diff
@@ -30,82 +30,29 @@ public static void AssertEquals(string expectedValue, DataFrameColumn arrowArray
...
+            };
+            if (arrowType == null)
+                throw new NotSupportedException($"Unknown type: {typeof(T)}");
+            return arrowType;
```
Comment (suhsteve) [inline src/csharp/Microsoft.Spark.UnitTest/TestUtils/ArrowTestUtils.cs:55]: remove

File: src/csharp/Microsoft.Spark.UnitTest/TestUtils/ArrowTestUtils.cs [hunk 4]
```diff
@@ -120,82 +67,29 @@ public static ArrowStringDataFrameColumn ToArrowStringDataFrameColumn(StringArra
...
-            }
-
-            if (typeof(T) == typeof(byte[]))
-            {
-                return ToBinaryArrowArray((byte[][])(object)array);
-            }
-
-            throw new NotSupportedException($"Unknown type: {typeof(T)}");
+            Type type = typeof(T);
+            IArrowArray arrowType = type switch
```
Comment (suhsteve) [inline src/csharp/Microsoft.Spark.UnitTest/TestUtils/ArrowTestUtils.cs:71]: ditto

File: src/csharp/Microsoft.Spark.UnitTest/TestUtils/ArrowTestUtils.cs [hunk 5]
```diff
@@ -120,82 +67,29 @@ public static ArrowStringDataFrameColumn ToArrowStringDataFrameColumn(StringArra
...
+                _ when type == typeof(uint) => ToPrimitiveArrowArray((uint[])(object)array),
+                _ when type == typeof(long) => ToPrimitiveArrowArray((long[])(object)array),
+                _ when type == typeof(ulong) => ToPrimitiveArrowArray((ulong[])(object)array),
+                _ when type == typeof(float) => ToPrimitiveArrowArray((float[])(object)array),
+                _ when type == typeof(double) => ToPrimitiveArrowArray((double[])(object)array),
+                _ when type == typeof(DateTime) => ToPrimitiveArrowArray((DateTime[])(object)array),
+                _ when type == typeof(TimeSpan) => ToPrimitiveArrowArray((TimeSpan[])(object)array),
+                _ when type == typeof(string) => ToStringArrowArray((string[])(object)array),
+                _ when type == typeof(byte[]) => ToBinaryArrowArray((byte[][])(object)array),
+                _ => null
```
Comment (suhsteve) [inline src/csharp/Microsoft.Spark.UnitTest/TestUtils/ArrowTestUtils.cs:88]: ditto

File: src/csharp/Microsoft.Spark.UnitTest/TestUtils/ArrowTestUtils.cs [hunk 6]
```diff
@@ -120,82 +67,29 @@ public static ArrowStringDataFrameColumn ToArrowStringDataFrameColumn(StringArra
...
+            };
+            if (arrowType == null)
+                throw new NotSupportedException($"Unknown type: {typeof(T)}");
+            return arrowType;
```
Comment (suhsteve) [inline src/csharp/Microsoft.Spark.UnitTest/TestUtils/ArrowTestUtils.cs:92]: ditto

File: src/csharp/Microsoft.Spark.UnitTest/TestUtils/ArrowTestUtils.cs [hunk 7]
```diff
@@ -120,82 +67,29 @@ public static ArrowStringDataFrameColumn ToArrowStringDataFrameColumn(StringArra
...
-                return ToStringArrowArray((string[])(object)array);
-            }
-
-            if (typeof(T) == typeof(byte[]))
-            {
-                return ToBinaryArrowArray((byte[][])(object)array);
-            }
-
-            throw new NotSupportedException($"Unknown type: {typeof(T)}");
+            Type type = typeof(T);
```
Comment (suhsteve) [inline src/csharp/Microsoft.Spark.UnitTest/TestUtils/ArrowTestUtils.cs:70]: Let's define `object arrayObject = array;` here and use it below.

**Comment by SARAVANA1501** [comment]
@suhsteve Changes done, Can you review again.?

File: src/csharp/Microsoft.Spark.UnitTest/TestUtils/ArrowTestUtils.cs (hunk 7 above)
Comment (suhsteve) [reply src/csharp/Microsoft.Spark.UnitTest/TestUtils/ArrowTestUtils.cs:67]: You didn't implement this suggestion.

Comment (SARAVANA1501) [reply src/csharp/Microsoft.Spark.UnitTest/TestUtils/ArrowTestUtils.cs:67]: I am unable to understand your suggestion, can you explain a little bit more? why we have define arrayObject and where it should be used?

Comment (suhsteve) [reply src/csharp/Microsoft.Spark.UnitTest/TestUtils/ArrowTestUtils.cs:67]: define `object arrayObject = array;` and replace `(object)array` with `arrayObject`

**Comment by SARAVANA1501** [comment]
@suhsteve Changes done, Can you review again.?

**Review by suhsteve — APPROVED** [review]
LGTM.  Thanks @SARAVANA1501 !

**Comment by SARAVANA1501** [comment]
@suhsteve when this will be merged? any release cycle?

**Comment by imback82** [comment]
@SARAVANA1501 Thanks for your patience. We have been busy with releasing 1.0, and we will get back to PRs soon.

**Comment by Niharikadutta** [comment]
LGTM. Thanks @SARAVANA1501 !

**Review by imback82 — APPROVED** [review]
LGTM, thanks @SARAVANA1501!

**Pull request merged**